tests/functional/v2/task/testIngestShards.js

```diff
@@ -144,5 +144,46 @@ describe('Test IngestShards', function () {
         );
         series[0].values.forEach(val => assert.strictEqual(val.id, undefined));
     });
+
+    it('should increment microseconds for duplicate timestamps', async () => {
```

Comment:
Do we test the case where there are >2 events conflicting on the same ts ?

Reply:
I'll add one